components/camel-arangodb/src/main/java/org/apache/camel/component/arangodb/ArangoDbComponent.java

```diff
@@ -18,15 +18,44 @@
 
 import java.util.Map;
 
+import org.apache.camel.CamelContext;
 import org.apache.camel.Endpoint;
+import org.apache.camel.spi.Metadata;
+import org.apache.camel.spi.annotations.Component;
 import org.apache.camel.support.DefaultComponent;
 
-@org.apache.camel.spi.annotations.Component("arangodb")
+@Component("arangodb")
 public class ArangoDbComponent extends DefaultComponent {
 
+    @Metadata
+    private ArangoDbConfiguration configuration = new ArangoDbConfiguration();
+
+    public ArangoDbComponent() {
+        this(null);
+    }
+
+    public ArangoDbComponent(CamelContext context) {
+        super(context);
+    }
+
     protected Endpoint createEndpoint(String uri, String remaining, Map<String, Object> parameters) throws Exception {
-        Endpoint endpoint = new ArangoDbEndpoint(uri, this);
+        final ArangoDbConfiguration configuration = this.configuration != null ? this.configuration.copy() : new ArangoDbConfiguration();
+        configuration.setDatabase(remaining);
```

Comment:
Perhaps before setting the `database`, it would make sense to check its validity (e.g: is not empty)

Reply:
since arango driver has default values, I don't see the need for the check here

Reply:
we do need database name at minimum for any operation